src/components/NewDatePicker/CalendarPicker/YearPickerModal.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../../withCurrentUserPersonalDetails';
+import HeaderWithBackButton from '../../HeaderWithBackButton';
+import withLocalize, {withLocalizePropTypes} from '../../withLocalize';
+import CONST from '../../../CONST';
+import SelectionListRadio from '../../SelectionListRadio';
+import Modal from '../../Modal';
+import {radioListItemPropTypes} from '../../SelectionListRadio/selectionListRadioPropTypes';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+    ...withCurrentUserPersonalDetailsPropTypes,
+
+    /** Whether the modal is visible */
+    isVisible: PropTypes.bool.isRequired,
+
+    /** The list of years to render */
+    years: PropTypes.arrayOf(PropTypes.shape(radioListItemPropTypes)).isRequired,
+
+    /** Currently selected year */
+    currentYear: PropTypes.number.isRequired,
+
+    /** Function to call when the user selects a year */
+    onYearChange: PropTypes.func.isRequired,
+
+    /** Value for the text input */
+    textInputValue: PropTypes.string.isRequired,
+
+    /** Callback to fire when the text input changes */
+    onChangeText: PropTypes.func.isRequired,
+
+    /** Function to call when the user closes the year picker */
+    onClose: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    ...withCurrentUserPersonalDetailsDefaultProps,
+    isVisible: false,
+    years: [],
+    currentYear: new Date().getFullYear(),
+    onYearChange: () => {},
+    textInputValue: '',
+    onChangeText: () => {},
+    onClose: () => {},
+};
+
+function YearPickerModal(props) {
+    const headerMessage = props.textInputValue.trim() && !props.years.length ? props.translate('common.noResultsFound') : '';
+
+    return (
+        <Modal
+            type={CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED}
+            isVisible={props.isVisible}
+            onClose={props.onClose}
+            onModalHide={props.onClose}
+            hideModalContentWhileAnimating
```

Comment:
Yep, `Modal` spreads the props:

```jsx
function Modal(props) {
    return (
        <BaseModal
            // eslint-disable-next-line react/jsx-props-no-spreading
            {...props}
        >
            {props.children}
        </BaseModal>
    );
}
```

and then it goes all the way down to `BaseModal`, which uses it:

```jsx
<ReactNativeModal
    hideModalContentWhileAnimating={this.props.hideModalContentWhileAnimating}
    ...
>
    ...
</ReactNativeModal>
```

`hideModalContentWhileAnimating` comes from `react-native-modal` and is used to prevent flashes in the UI when `useNativeDriver` is true. 

See here:
https://github.com/react-native-modal/react-native-modal/issues/343#issuecomment-520243962

and here:
https://github.com/react-native-modal/react-native-modal/pull/116

Reply:
@fedirjh Yeah, seems like there are a few other props missing too. I could add them as part of this PR, do you think it makes sense?